packages/blocks/src/api/utils.js

```diff
@@ -121,3 +123,105 @@ export function normalizeBlockType( blockTypeOrName ) {
 
 	return blockTypeOrName;
 }
+
+/**
+ * Get the label for the block, usually this is either the block title,
+ * or the value of the block's `label` function when that's specified.
+ *
+ * @param {Object} blockType  The block type.
+ * @param {Object} attributes The values of the block's attributes.
+ * @param {Object} context    The intended use for the label.
```

Comment:
This isn't an object. If we documented this as an union type of string literals, I think it would have helped clarify (both for us as the implementers and the future consumers) which sorts of values we're expecting to be supported.

```suggestion
 * @param {"visual"|"accessibility"} context The intended use for the label
```

It is a bit confusing for me though in how we expect block implementers to be handling this. Even in the implementations we've revised so far, we're only ever handling the "accessibility" case. So what's the point of "visual"? And how would we expect to treat the `undefined` return value from those implementations? Should have gone ahead and returned a value for the "visual" context for existing block revisions? (Getting a bit of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) vibes here)

Reply:
I'm now working on having two separate params in the block definition, so this won't be an issue any longer.

Reply:
> I'm now working on having two separate params in the block definition, so this won't be an issue any longer.

What about the second part of my comment, in which I question whether we even need two?

>It is a bit confusing for me though in how we expect block implementers to be handling this. Even in the implementations we've revised so far, we're only ever handling the "accessibility" case. So what's the point of "visual"? And how would we expect to treat the undefined return value from those implementations? Should have gone ahead and returned a value for the "visual" context for existing block revisions? (Getting a bit of YAGNI vibes here)